src/cloud-api-adaptor/test/e2e/assessment_runner.go

```diff
@@ -437,16 +436,28 @@ func (tc *TestCase) Run() {
 					} else if profile != "" {
 						fmt.Printf("PodVM Created with Instance Type %v", profile)
 						if tc.FailReason != "" {
-							jsonData, err := json.Marshal(tc.pod.Status)
-							if err != nil {
-								fmt.Println("Error marshaling pod.Status to JSON: ", err.Error())
-							} else {
-								t.Logf("Current Pod State: %v", string(jsonData))
+							var podlist v1.PodList
+							var podLogString string
+							if err := client.Resources("confidential-containers-system").List(ctx, &podlist); err != nil {
+								t.Fatal(err)
+							}
+							for _, pod := range podlist.Items {
+								if pod.Labels["app"] == "cloud-api-adaptor" {
+									podLogString, _ = GetPodLog(ctx, client, pod)
+									break
+								}
 							}
-							if strings.Contains(string(jsonData), tc.FailReason) {
+							if strings.Contains(podLogString, tc.FailReason) {
 								log.Infof("failed due to expected reason %s", tc.FailReason)
 							} else {
-								t.Fatal(err)
+								log.Infof("Here is the podLog %s", podLogString)
```

Comment:
Still wording tweak for clarity:
```suggestion
								log.Infof("cloud-api-adaptor pod logs: %s", podLogString)
```